scripts/validate_string_concatenation.py

```diff
@@ -102,28 +106,157 @@ def strings_to_concatenate(source_path: str) -> Generator[Tuple[str, int], None,
         Source file path.
     line_number : int
         Line number of unconcatenated string.
+    MSG : str
+        Explenation of the error.
+
+    Notes
+    -----
+    GH #30454
     """
+    MSG: str = (
+        "String unnecessarily split in two by black. Please merge them manually."
+    )
+
     with open(source_path, "r") as file_name:
         tokens: List = list(tokenize.generate_tokens(file_name.readline))
 
     for current_token, next_token in zip(tokens, tokens[1:]):
         if current_token[0] == next_token[0] == token.STRING:
-            yield source_path, current_token[2][0]
+            yield source_path, current_token[2][0], MSG
+
+
+def strings_with_wrong_placed_space(
+    source_path: str,
+) -> Generator[Tuple[str, int, str], None, None]:
+    """
+    Test case for leading spaces in concated strings.
+
+    For example:
+
+    >>> foo = (
+    ...    "bar "
+    ...    "baz"
+    ... )
+
+    Instead of:
+
+    >>> foo = (
+    ...        "bar"
+    ...       " baz"
+    ... )
+
+    Parameters
+    ----------
+    source_path : str
+        File path pointing to a single file.
+
+    Yields
+    ------
+    source_path : str
+        Source file path.
+    line_number : int
+        Line number of unconcatenated string.
+    MSG : str
+        Explenation of the error.
+    """
+    MSG: str = (
+        "String has a space at the beginning "
+        "instead of the end of the previous string."
+    )
+
+    with open(source_path, "r") as file_name:
+        tokens: List = list(tokenize.generate_tokens(file_name.readline))
+
+    for first_token, second_token, third_token in zip(tokens, tokens[1:], tokens[2:]):
+        if (
+            first_token[0] == third_token[0] == token.STRING
+            and second_token[0] == token.NL
+        ):
+            # Means we are in a block of concated string
+
+            # Striping the quotes
+            first_string = first_token[1][1:-1]
+            second_string = third_token[1][1:-1]
+
+            if (not first_string.endswith(" ")) and (second_string.startswith(" ")):
+                yield source_path, third_token[2][0], MSG
+
+
+def bare_pytest_raises(source_path: str) -> Generator[Tuple[str, int, str], None, None]:
+    """
+    Test Case for bare pytest raises.
+
+    For example:
+
+    >>> with pytest.raise(ValueError):
+    ...     # Some code that raises ValueError
+
+    Instead of:
```

Comment:
```suggestion
    And this is what we want instead:
```

Otherwise, it's difficult to understand which is the good one or not (I'd guess it's the opposite if I didn't know.

Reply:
Applied.